lib/matplotlib/artist.py

```diff
@@ -845,21 +845,46 @@ def update(self, props):
         """
         store = self.eventson
         self.eventson = False
-        changed = False
+        try:
+            ret = [self._update_property(k, v)
+                   for k, v in sorted(props.items(), reverse=True)]
```

Comment:
Partly. It is also to have a priority structure. If someone defines both
"color" and "facecolor" at the same time, the current behavior is for color
to take precedence over facecolor, IIRC.

On Tue, Dec 1, 2015 at 11:51 AM, Michael Droettboom <
notifications@github.com> wrote:

> In lib/matplotlib/artist.py
> <https://github.com/matplotlib/matplotlib/pull/5599#discussion_r46304017>:
>
> > @@ -845,21 +845,46 @@ def update(self, props):
> >          """
> >          store = self.eventson
> >          self.eventson = False
> > -        changed = False
> > +        try:
> > +            ret = [self._update_property(k, v)
> > +                   for k, v in sorted(props.items(), reverse=True)]
>
> The sorting by key name is just to prevent any non-deterministic behavior?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5599/files#r46304017>.
>